view/adminhtml/templates/store_select.phtml

```diff
@@ -89,13 +93,22 @@ $block = $block;
       minimumResultsForSearch: Infinity, // disable search
       width: '25%'
     });
+
     $("#ns8-store-select").on("change", function() {
       var storeId = this.value.toString();
       var store = stores.find(function(store) {
         return store.id === storeId;
       });
-      renderProtectIFrame(store.token);
+      if(storeId !== paramStoreId) {
+        const newUrl = '<?=$block->getUrl('*/*/*')?>' + '?store_id=' + storeId;
+        window.location = newUrl;
+      }else{
+        renderProtectIFrame(store.token);
+      }
```

Comment:
Non-blocking:

- Need a space after `if` and before `(`
- Need spaces around `else`